File: backend/UserManager/src/main/java/hu/bmester/usermanager/dto/NewUserDTO.java
package hu.bmester.usermanager.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;

@Data
public class NewUserDTO {

    @NotBlank
    private String lastName;

    @NotBlank
    private String firstName;

    @NotBlank
    private String birthName;

    private String mothersName;

    private Long genderId;

    @NotNull
    private Long nationalityId;

    @NotBlank
    private String phone;

    @NotBlank
    private String taxNumber;

    @NotNull
    List<NewAddressDTO> addresses;

}

File: frontend/src/app/dto/new-user-dto.ts
import {Gender} from "../model/gender";
import {Nationality} from "../model/nationality";
import {NewAddressDTO} from "./new-address-dto";

export class NewUserDTO {
  lastName?: String;
  firstName?: String;
  birthName?: String;
  mothersName?: String;
  genderId?: number;
  nationalityId?: number;
  addresses?: NewAddressDTO[];
  phone?: String;
  taxNumber?: String;
  note?: String;
}
